File: БР1.2/Smirnova_Karina/labs/lab2/src/main/java/org/renting/rentingservice/controller/BookingController.java
package org.renting.rentingservice.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.renting.rentingservice.domain.enums.BookingStatus;
import org.renting.rentingservice.dto.booking.BookingResponse;
import org.renting.rentingservice.dto.booking.CreateBookingRequest;
import org.renting.rentingservice.dto.common.PageResponse;
import org.renting.rentingservice.security.SecurityUtils;
import org.renting.rentingservice.service.BookingService;
import org.springframework.data.domain.Pageable;
import org.springframework.data.web.PageableDefault;
import org.springframework.http.HttpStatus;
import org.springframework.context.annotation.Profile;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@Profile("property")
@RequestMapping("/bookings")
@Tag(name = "Bookings", description = "Бронирования жилья")
public class BookingController {

    private final BookingService bookingService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    @Operation(summary = "Создать бронирование", description = "Создаёт новое бронирование для выбранного объявления")
    public BookingResponse create(@Valid @RequestBody CreateBookingRequest request) {
        return bookingService.create(SecurityUtils.currentUserId(), request);
    }

    @GetMapping
    @Operation(summary = "Список бронирований", description = "Возвращает бронирования текущего гостя с фильтром по статусу")
    public PageResponse<BookingResponse> list(
            @RequestParam(required = false) BookingStatus status,
            @PageableDefault(size = 20) Pageable pageable) {
        return bookingService.listForGuest(SecurityUtils.currentUserId(), status, pageable);
    }

    @GetMapping("/{bookingId}")
    @Operation(summary = "Детали бронирования", description = "Возвращает бронирование по идентификатору")
    public BookingResponse get(@PathVariable Long bookingId) {
        return bookingService.get(bookingId, SecurityUtils.currentUserId());
    }

    @PostMapping("/{bookingId}/cancel")
    @Operation(summary = "Отменить бронирование", description = "Отменяет бронирование текущим пользователем")
    public BookingResponse cancel(@PathVariable Long bookingId) {
        return bookingService.cancel(bookingId, SecurityUtils.currentUserId());
    }

    @PostMapping("/{bookingId}/complete")
    @Operation(summary = "Завершить бронирование", description = "Помечает бронирование как завершённое")
    public BookingResponse complete(@PathVariable Long bookingId) {
        return bookingService.complete(bookingId, SecurityUtils.currentUserId());
    }
}

File: БР1.2/Цатинян Артём/labs/lab1/src/main/kotlin/ru/itmo/restaurantbooking/booking/adapter/rest/BookingController.kt
package ru.itmo.restaurantbooking.booking.adapter.rest

import io.swagger.v3.oas.annotations.security.SecurityRequirement
import jakarta.validation.Valid
import org.springdoc.core.annotations.ParameterObject
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.itmo.restaurantbooking.auth.domain.AuthenticatedUser
import ru.itmo.restaurantbooking.booking.adapter.rest.dto.BookingSearchRequest
import ru.itmo.restaurantbooking.booking.adapter.rest.dto.CreateBookingRequest
import ru.itmo.restaurantbooking.booking.adapter.rest.mapper.BookingMapper
import ru.itmo.restaurantbooking.common.adapter.rest.config.OpenApiConfig
import ru.itmo.restaurantbooking.booking.service.BookingService

@RestController
@RequestMapping("/api/v1/bookings")
@SecurityRequirement(name = OpenApiConfig.BEARER_AUTH)
class BookingController(
    private val bookingService: BookingService,
    private val bookingMapper: BookingMapper
) {
    @PostMapping
    fun create(
        authentication: Authentication,
        @Valid @RequestBody request: CreateBookingRequest
    ) = bookingService.create(
            userId = authentication.userId(),
            request = request
        )

    @GetMapping("/me")
    fun getMine(
        authentication: Authentication,
        @ParameterObject @Valid @ModelAttribute request: BookingSearchRequest
    ) = bookingService.getMine(
            userId = authentication.userId(),
            query = bookingMapper.toSearchQuery(request)
        )

    @GetMapping("/{bookingId}")
    fun getById(
        authentication: Authentication,
        @PathVariable bookingId: Long
    ) = bookingService.getById(
            userId = authentication.userId(),
            bookingId = bookingId
        )

    @PatchMapping("/{bookingId}/cancel")
    fun cancel(
        authentication: Authentication,
        @PathVariable bookingId: Long
    ) = bookingService.cancel(
            userId = authentication.userId(),
            bookingId = bookingId
        )

    private fun Authentication.userId() = (principal as AuthenticatedUser).id
}
